src/main/java/net/geant/nmaas/portal/api/bulk/BulkController.java

```diff
@@ -114,11 +117,31 @@ private List<BulkDeploymentViewS> mapToView(List<BulkDeployment> deployments) {
                 .map(bulk -> {
                     BulkDeploymentViewS bulkView = modelMapper.map(bulk, BulkDeploymentViewS.class);
                     bulkView.setCreator(getUserView(bulk.getCreatorId()));
+                    mapDetails(bulk, bulkView);
                     return bulkView;
                 })
                 .collect(Collectors.toList());
     }
 
+    private BulkDeploymentViewS mapDetails(BulkDeployment deployment, BulkDeploymentViewS view) {
+        if(deployment.getType().equals(BulkType.APPLICATION)) {
+            Map<String, String> details = new HashMap<>();
+            if(!deployment.getEntries().isEmpty()) {
+                details.put(BulkDeploymentViewS.BULK_ENTRY_DETAIL_KEY_APP_INSTANCE_NO, String.valueOf(deployment.getEntries().size()));
+                if(deployment.getEntries().stream().findFirst().get().getDetails().containsKey(BulkDeploymentEntryView.BULK_ENTRY_DETAIL_KEY_APP_ID)) {
```

Comment:
Save this first entry to a local variable in order not to repeat this long chain multiple times